packages/sanity/src/core/store/_legacy/user/userStore.ts

```diff
@@ -74,9 +74,11 @@ export function createUserStore({client: _client, currentUser}: UserStoreOptions
 
   userLoader.prime('me', userFromCurrentUser)
 
-  if (userFromCurrentUser?.id) {
+  //@todo figure out if we are ok with this removal
```

Comment:
TODO/FIXME/COMMIT TO THIS:

Currently, `useUser(userId)` will return a `User` with `sanityUserId` (global user id), _unless_ the `userId` happens to be the current user id. This is surprising.

The reason is this line of code, which puts `currentUser` into the user cache. `currentUser` obtained from `/me` does NOT have all propereties obtained using the `/user/:id` endpoint.

I make the case here that we should just delete this line, and allow the extra request, so that global token for current user can be obtained like so:

```ts
const currentUser = useCurrentUser()
const userId = currentUser?.id ?? ''
const [user] = useUser(userId)
// user.sanityUserId
```

If there are good reasons not to do this, I can restore this line, and add some bespoke `/user/currentUserId` fetch in the integration codepath that needs it.

Reply:
Update: We found issues with this approach and will remove the "global-id-in-url" approach.
As it turns out the user can log in with a different user in Create, resulting in the link-creation failing. Instead Create will handle all this, so we no longer need global user id in Studio.

TLDR: all code related to user.id will be reverted/removed once Create has been updated with this new behavior.

Reply:
Reverted.